From ca39b05e0c8e4085960a6055ffbcd842abca84c3 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 10 Apr 2024 12:01:39 +0200
Subject: [PATCH 0757/1695] drivers: regulator: split
 regulator_supported_voltages() assertions

Split assertions in regulator_supported_voltages() to ease debugging,
providing a better indication of which condition is not fulfilled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5cc70df3c3a169dea0f61a4c258021650eb362bc drivers: regulator: split regulator_supported_voltages() assertions]
Change-Id: I349e0748d906105a02540dd6f7f7549450ba302f
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372700
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/regulator/regulator.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/core/drivers/regulator/regulator.c b/core/drivers/regulator/regulator.c
index 477b8c158..67d3c98de 100644
--- a/core/drivers/regulator/regulator.c
+++ b/core/drivers/regulator/regulator.c
@@ -263,12 +263,16 @@ TEE_Result regulator_supported_voltages(struct regulator *regulator,
 		return res;
 	}
 
-	assert(((*desc)->type == VOLTAGE_TYPE_FULL_LIST &&
-		(*levels)[0] >= regulator->min_uv && (*desc)->num_levels &&
-		(*levels)[(*desc)->num_levels - 1] <= regulator->max_uv) ||
-	       ((*desc)->type == VOLTAGE_TYPE_INCREMENT &&
-		(*levels)[0] >= regulator->min_uv &&
-		(*levels)[1] <= regulator->max_uv));
+	if ((*desc)->type == VOLTAGE_TYPE_FULL_LIST) {
+		assert((*desc)->num_levels);
+		assert((*levels)[0] >= regulator->min_uv);
+		assert((*levels)[(*desc)->num_levels - 1] <= regulator->max_uv);
+	} else if ((*desc)->type == VOLTAGE_TYPE_INCREMENT) {
+		assert((*levels)[0] >= regulator->min_uv);
+		assert((*levels)[1] <= regulator->max_uv);
+	} else {
+		assert(0);
+	}
 
 	return TEE_SUCCESS;
 }
-- 
2.47.2

